GoogleForCreators/web-stories-wp
# Shopping: Limit number of products per page

## Context



## Summary

Only allow 6 products on a page. 

## Relevant Technical Choices

Added some unit tests. 
Improved storybook. 
Add karma test. 

## To-do



## User-facing changes


https://user-images.githubusercontent.com/237508/174128271-593b0fee-1fcd-49b3-9136-132fcc7dc71c.mov


## Testing Instructions






This PR can be tested by following these steps:

#### Test 7 products insert. 
1. Create a story. 
2. Go to product tab. 
3. Insert 7 products. 
4. 7th product should not insert and message shown. 

#### Test copy and paste
1. Create a story. 
2. Go to product tab. 
3. Insert 5 products. 
4. Select and copy product elements. 
5. Create new page. 
6. Paste elements. 
7. Message should be shown and ensure that no more than 6 products are shown. 


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #11254
